bindings/python/tools/python/ostk/astrodynamics/pytrajectory/pystate.py

```
@@ -108,8 +124,21 @@
-    if all(column in data for column in eci_columns):
-        frame = Frame.GCRF()
+    match_groups: list[re.Match] = [
+        re.match(CANONICAL_FORMAT, column) for column in data.keys()
+    ]
+
+    if len(matches := [match for match in match_groups if match is not None]) == 6:
+        frame_name: str = matches[0].group(2)
+        try:
+            frame: Frame = Frame.with_name(frame_name) or getattr(Frame, frame_name)()
+        except Exception:
+            raise ValueError(f"No frame exists with name [{frame_name}].")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)